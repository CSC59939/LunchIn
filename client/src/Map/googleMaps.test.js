import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import GoogleM from "./googleMaps";

describe("Maps Component", () => {
  describe("When provided with no info", () => {
    it("Should render correctly", () => {
      const mapInstance = shallow(
        <GoogleM />
      );
      expect(mapInstance).toMatchSnapshot();
    });
  });
  describe("When provided with food and address info", () => {
    it("Should render correctly", () => {
      const food = "Pizza";
      const address = "160 Convent Ave, New York, NY 10031"
      const lat = 40.8216144
      const lng = -73.9479427
      const storeN = "Uncle Tony's Pizza"
      const storeL = "1596 Amsterdam Ave, New York"
      const mapInstance = shallow(
        <GoogleM
            food={food}
            address={address}
            lat={lat}
            lng={lng}
            storeN={storeN}
            storeL={storeL}
        />
      );
      expect(mapInstance).toMatchSnapshot();
    });
  });
});
