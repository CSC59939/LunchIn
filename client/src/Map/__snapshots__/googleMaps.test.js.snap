// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps Component When provided only lat and lng and address Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleMapComponent
    address="160 Convent Ave, New York, NY 10031"
    lat={40.8216144}
    lng={-73.9479427}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(CssBaseline) />,
        <WithStyles(Card)>
          <WithStyles(CssBaseline) />
          <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
            address="160 Convent Ave, New York, NY 10031"
            lat={40.8216144}
            lng={-73.9479427}
          />
        </WithStyles(Card)>,
        <WithStyles(Card)>
          <WithStyles(CssBaseline) />
          <BusinessCard />
        </WithStyles(Card)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CssBaseline) />,
            <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
              address="160 Convent Ave, New York, NY 10031"
              lat={40.8216144}
              lng={-73.9479427}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "address": "160 Convent Ave, New York, NY 10031",
              "lat": 40.8216144,
              "lng": -73.9479427,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CssBaseline) />,
            <BusinessCard />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "address": undefined,
              "storeLat": undefined,
              "storeLng": undefined,
              "storeName": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(CssBaseline) />,
          <WithStyles(Card)>
            <WithStyles(CssBaseline) />
            <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
              address="160 Convent Ave, New York, NY 10031"
              lat={40.8216144}
              lng={-73.9479427}
            />
          </WithStyles(Card)>,
          <WithStyles(Card)>
            <WithStyles(CssBaseline) />
            <BusinessCard />
          </WithStyles(Card)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CssBaseline) />,
              <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
                address="160 Convent Ave, New York, NY 10031"
                lat={40.8216144}
                lng={-73.9479427}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "address": "160 Convent Ave, New York, NY 10031",
                "lat": 40.8216144,
                "lng": -73.9479427,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CssBaseline) />,
              <BusinessCard />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "address": undefined,
                "storeLat": undefined,
                "storeLng": undefined,
                "storeName": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Maps Component When provided with food and address info Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleMapComponent
    address="160 Convent Ave, New York, NY 10031"
    food="Pizza"
    lat={40.8216144}
    lng={-73.9479427}
    storeL="1596 Amsterdam Ave, New York"
    storeN="Uncle Tony's Pizza"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(CssBaseline) />,
        <WithStyles(Card)>
          <WithStyles(CssBaseline) />
          <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
            address="160 Convent Ave, New York, NY 10031"
            food="Pizza"
            lat={40.8216144}
            lng={-73.9479427}
            updateMarker={[Function]}
          />
        </WithStyles(Card)>,
        <WithStyles(Card)>
          <WithStyles(CssBaseline) />
          <BusinessCard />
        </WithStyles(Card)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CssBaseline) />,
            <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
              address="160 Convent Ave, New York, NY 10031"
              food="Pizza"
              lat={40.8216144}
              lng={-73.9479427}
              updateMarker={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Pizza",
            "nodeType": "function",
            "props": Object {
              "address": "160 Convent Ave, New York, NY 10031",
              "food": "Pizza",
              "lat": 40.8216144,
              "lng": -73.9479427,
              "updateMarker": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CssBaseline) />,
            <BusinessCard />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "address": undefined,
              "storeLat": undefined,
              "storeLng": undefined,
              "storeName": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(CssBaseline) />,
          <WithStyles(Card)>
            <WithStyles(CssBaseline) />
            <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
              address="160 Convent Ave, New York, NY 10031"
              food="Pizza"
              lat={40.8216144}
              lng={-73.9479427}
              updateMarker={[Function]}
            />
          </WithStyles(Card)>,
          <WithStyles(Card)>
            <WithStyles(CssBaseline) />
            <BusinessCard />
          </WithStyles(Card)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CssBaseline) />,
              <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
                address="160 Convent Ave, New York, NY 10031"
                food="Pizza"
                lat={40.8216144}
                lng={-73.9479427}
                updateMarker={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Pizza",
              "nodeType": "function",
              "props": Object {
                "address": "160 Convent Ave, New York, NY 10031",
                "food": "Pizza",
                "lat": 40.8216144,
                "lng": -73.9479427,
                "updateMarker": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CssBaseline) />,
              <BusinessCard />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "address": undefined,
                "storeLat": undefined,
                "storeLng": undefined,
                "storeName": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Maps Component When provided with no info Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleMapComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(CssBaseline) />,
        <WithStyles(Card)>
          <WithStyles(CssBaseline) />
          <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component)))))))) />
        </WithStyles(Card)>,
        <WithStyles(Card)>
          <WithStyles(CssBaseline) />
          <BusinessCard />
        </WithStyles(Card)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CssBaseline) />,
            <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component)))))))) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "address": undefined,
              "lat": undefined,
              "lng": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CssBaseline) />,
            <BusinessCard />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "address": undefined,
              "storeLat": undefined,
              "storeLng": undefined,
              "storeName": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(CssBaseline) />,
          <WithStyles(Card)>
            <WithStyles(CssBaseline) />
            <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component)))))))) />
          </WithStyles(Card)>,
          <WithStyles(Card)>
            <WithStyles(CssBaseline) />
            <BusinessCard />
          </WithStyles(Card)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CssBaseline) />,
              <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component)))))))) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "address": undefined,
                "lat": undefined,
                "lng": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CssBaseline) />,
              <BusinessCard />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "address": undefined,
                "storeLat": undefined,
                "storeLng": undefined,
                "storeName": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
